File: docs/legacy(old PACE)/unsupported/VB.NET/PaceWebServiceSDK/sample/FindObjects.vb
Imports PaceWebServiceSDK.efipaceservices

Public Class FindObjects
    Public Shared Sub Run()
        Dim findObjectsHttpBinding As FindObjectsHttpBinding = PaceClient.getFindObjectsHttpBinding()

        'create a sort for the keys
        Dim sort As XPathDataSort() = New XPathDataSort(2) {}

        Dim xPathDataSort1 As XPathDataSort = New XPathDataSort()
        xPathDataSort1.descending = False
        xPathDataSort1.xpath = "customer/@custName"

        Dim xPathDataSort2 As XPathDataSort = New XPathDataSort()

        xPathDataSort2.descending = True
        xPathDataSort2.xpath = "@description"

        sort(0) = xPathDataSort1
        sort(1) = xPathDataSort2

        'Open Job Filter Sample
        Dim keys As String() = findObjectsHttpBinding.find("Job", "adminStatus/@openJob")
        Console.WriteLine(keys.Length.ToString + " Open Jobs")
        Console.WriteLine(String.Join(",", keys))
    End Sub

End Class

File: docs/legacy(old PACE)/C#/PaceAPISample/sample/FindObjects.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;

namespace Pace_Web_Service_SDK.sample
{
    class FindObjects
    {
        public static void Run()
        {
            FindObjectsPortType findObjectsHttpBinding = PaceClient.getFindObjectsHttpBinding();

            // create a sort for the keys
            XPathDataSort[] sort = new XPathDataSort[2];

            XPathDataSort xPathDataSort1 = new XPathDataSort();
            xPathDataSort1.descending = false;
            xPathDataSort1.xpath = "customer/@custName";

            XPathDataSort xPathDataSort2 = new XPathDataSort();
            xPathDataSort2.descending = true;
            xPathDataSort2.xpath = "@description";

            sort[0] = xPathDataSort1;
            sort[1] = xPathDataSort2;

            //Open Job Filter Sample
            ArrayOfString keys = findObjectsHttpBinding.findAndSort("Job", "adminStatus/@openJob", sort);

            //Date Filter Sample
            ArrayOfString keys2 = findObjectsHttpBinding.find("Job", "@dateSetup = date( 2012, 07, 03 )");

            Console.WriteLine(keys2.Count + " Jobs Setup As Of 2012-07-03");

            //Finds ALL Job Shipments for a particular job, part, with a not null tracking number
            ArrayOfString shipments = findObjectsHttpBinding.find("JobShipment", " ( @job ='090005' and @jobPart ='01' ) and @trackingNumber !=''");

            foreach (String shipment in shipments)
            {
                Console.WriteLine(shipment);
            }
        }
    }
}
